File: scripts/build-media-manifest.js
#!/usr/bin/env node
// Scans media/ for the favicon/social-share image pools and records them in
// content.json's `site_images` key. Re-run after adding new files named
// `icon-*` (favicon candidates, ideally square) or `ogg-*` (social share
// image candidates, ideally ~1200x630+) to media/, then redeploy.
'use strict';
const fs = require('fs');
const path = require('path');

const REPO = path.dirname(__dirname);
const MEDIA_DIR = path.join(REPO, 'media');
const CONTENT = process.argv[2] || path.join(REPO, 'content.json');

const EXT_RE = /\.(png|jpe?g|webp)$/i;

function pool(prefix) {
  return fs.readdirSync(MEDIA_DIR)
    .filter((f) => f.toLowerCase().startsWith(prefix) && EXT_RE.test(f))
    .sort()
    .map((f) => `media/${f}`);
}

function main() {
  const data = JSON.parse(fs.readFileSync(CONTENT, 'utf8'));
  data.site_images = {
    icons: pool('icon-'),
    ogg: pool('ogg-'),
  };
  fs.writeFileSync(CONTENT, JSON.stringify(data, null, 2));

  console.log(`icons: ${data.site_images.icons.length}`);
  data.site_images.icons.forEach((p) => console.log(`  ${p}`));
  console.log(`ogg: ${data.site_images.ogg.length}`);
  data.site_images.ogg.forEach((p) => console.log(`  ${p}`));
}

main();

File: scripts/build_media_manifest.py
#!/usr/bin/env python3
"""Scans media/ for the favicon/social-share image pools and records them in
content.json's `site_images` key. Re-run this after adding new files named
`icon-*` (favicon candidates, ideally square) or `ogg-*` (social share image
candidates, ideally ~1200x630+) to media/, then redeploy."""
import json
import os
import re
import sys

REPO = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
MEDIA_DIR = os.path.join(REPO, 'media')
CONTENT = sys.argv[1] if len(sys.argv) > 1 else os.path.join(REPO, 'content.json')

EXT_RE = re.compile(r'\.(png|jpe?g|webp)$', re.I)


def pool(prefix):
    files = [f for f in os.listdir(MEDIA_DIR) if f.lower().startswith(prefix) and EXT_RE.search(f)]
    return sorted(f"media/{f}" for f in files)


def main():
    with open(CONTENT) as f:
        data = json.load(f)

    data['site_images'] = {
        "icons": pool('icon-'),
        "ogg": pool('ogg-'),
    }

    with open(CONTENT, 'w') as f:
        json.dump(data, f, indent=2)

    print(f"icons: {len(data['site_images']['icons'])}")
    for p in data['site_images']['icons']:
        print(f"  {p}")
    print(f"ogg: {len(data['site_images']['ogg'])}")
    for p in data['site_images']['ogg']:
        print(f"  {p}")


if __name__ == '__main__':
    main()
